### ISSUE TITLE
Optional boolean resolves to true, false or undefiend

### ISSUE THREAD

**Issue by avin-kavish** [Hookyns/tst-reflect#60]
```
class Foo {
  boolField? = false
}
```
This resolves to a container type with `undefined | true | false` so `type.isBoolean()` check on either the `true` type or the `false` type fails.

```
[
  TypeActivator {
    _name: 'false',
    _fullName: 'false',
    _kind: 2,
    _constructors: [],
    _properties: [],
    _indexes: [],
    _methods: [],
    _decorators: [],
    _typeParameters: [],
    _ctor: undefined,
    _ctorDesc: ConstructorImportActivator {},
    _interface: undefined,
    _isUnion: false,
    _isIntersection: false,
    _types: [],
    _literalValue: undefined,
    _typeArgs: [],
    _conditionalType: undefined,
    _indexedAccessType: undefined,
    _functionType: undefined,
    _genericTypeConstraint: undefined,
    _genericTypeDefault: undefined,
    _isGenericType: false,
    _genericTypeDefinition: undefined,
    _baseType: LazyType {
      typeResolver: [Function (anonymous)],
      resolvedType: [TypeActivator]
    }
  },
  TypeActivator {
    _name: 'true',
    _fullName: 'true',
    _kind: 2,
    _constructors: [],
    _properties: [],
    _indexes: [],
    _methods: [],
    _decorators: [],
    _typeParameters: [],
    _ctor: undefined,
    _ctorDesc: ConstructorImportActivator {},
    _interface: undefined,
    _isUnion: false,
    _isIntersection: false,
    _types: [],
    _literalValue: undefined,
    _typeArgs: [],
    _conditionalType: undefined,
    _indexedAccessType: undefined,
    _functionType: undefined,
    _genericTypeConstraint: undefined,
    _genericTypeDefault: undefined,
    _isGenericType: false,
    _genericTypeDefinition: undefined,
    _baseType: LazyType {
      typeResolver: [Function (anonymous)],
      resolvedType: [TypeActivator]
    }
  }
]
```

**Comment by Hookyns** [comment]
What's your tsconfig?

I get
```typescript
_ßr.Type.store.set(11356, {
	k: 1,
	isg: false,
	n: "Foo",
	fn: "@@this/dev/quick-tests/index.ts:Foo#11356",
	props: [{
		n: "boolField",
		t: _ßr.Type.store.wrap({
			n: "boolean",
			k: 2,
			ctor: function () {
				return Promise.resolve(Boolean);
			}
		}),
		am: 2,
		acs: 0,
		ro: false,
		o: true
	}],
	indxs: [],
	args: [],
	ctors: [{params: []}],
	ctor: function () {
		return Promise.resolve(Foo);
	}
});
```

Which is optional boolean property, as it should be.

**Comment by Hookyns** [comment]
Okay, I found it. That's cuz of `strictNullChecks` option.
That's related to #45

**Comment by avin-kavish** [comment]
I have that turned on. Do you mean `exactOptionalPropertyTypes`?

**Comment by Hookyns** [comment]
The behavior of transformer is wrong and is caused by the `strictNullChecks` option.
I've already prepared a fix. Right now, I'm looking into the #45, which is the same problem but from another side.

**Comment by Hookyns** [comment]
Well I've found some issues with that fix. I've found super simple solution inside TypeScript which just take type and returns original type without "optionality", but it strip off `null`s too. Different solution will be complicated but I'm working on it.

**Comment by Hookyns** [comment]
> I wrote this message a day while working on the solution so I changed my mind several times while progressing.. I keep that so you can see (maybe 😄) why I did what I did in the end.

After some digging I found out that counting with `true | false` will be necessary.

I've prepared a solution of this case, so with `strictNullChecks`, there is no `undefined` if the property is optional and there is only `boolean`. So that's what u wanted.

But if you do something like this:
```typescript
class Foo {
	boolField?: boolean | null;
}
```
the result will be `null | true | false`. That's just how TypeScript handle that, it's another simplification. Because Boolean is just an union of `true` and `false` literals. TypeScript creates one union with all the values rather than union of unions.

We have to count with `true` and `false` literals anyway. Image type `string | true` or `"true" | true`. You must be able to work with that. That's not boolean, it's only `true`.

Final thought... We should not remove that `undefined` from optional things. Imagine that you reflect over method. You want to know what argument it accepts so you will take the first param, take the type of that param and pass it to another method which should do something with that type, eg. resolve some value which can be assigned into that type. If we remove the `undefined` from that type, we actualy change the type and disallow the `undefined` value. 
And `Type.Undefined.isAssignableTo(typeOfThatParameter)` will be false which is wrong IMHO.

Solution:
I keep those unions with `null` and `undefined` but I did (de)simplification of union `true | false` into `boolean`. Sou optional method parameters and optional properties will be always unions with `undefined`.

Also I added `Type.isTrue()`and `Type.isFalse()`.
`getType<true>().isTrue()` returns true. 
`getType<true>().isBoolean()` returns false. 
`getType<true>().isAssignableTo(Type.Boolean)` returns true.

Published versions: `tst-reflect-transformer@0.11.3`, `tst-reflect@0.11.2`.
